boardgame/BoardGameImpl.java

```diff
@@ -0,0 +1,48 @@
+package boardgame;
+
+public class BoardGameImpl implements IBoardGame {
+    private Boolean[][] boardSlot = new Boolean[3][3];
+    // winners are (3,0) (0,3) (3,6) (6,3) (3,3)
+    private int[] sumOfX = {0, 0};
+    private int[] sumOfO = {0, 0};
+    private int xTurnNumber = 1;
+    private int oTurnNumber = 1;
+
+    @Override
+    public String GetWinner(){
+        if(this.xTurnNumber > 3 && this.sumOfX[0] % 3 == 0 && this.sumOfX[1] % 3 == 0){
+            return "X";
+        }
+        else if(this.oTurnNumber > 3 && this.sumOfO[0] % 3 == 0 && this.sumOfO[1] % 3 == 0){
+            return "O";
+        }
+        return null;
+    }
+
+    @Override
+    public boolean TakeSlot(boolean isX, int row, int column){
+        if(this.boardSlot[row][column] == null){
+            this.boardSlot[row][column] = isX;
+            addValueOfChecker(isX, row, column);
+            return true;
+        }
+        return false;
+    }
+
+    public void addValueOfChecker(boolean isX,int row, int column){
+        if(isX){
```

Comment:
โค้ดใน if..else มันเหมือนกันเดี๊ยเลย ต่างกันแค่เลือกใช้คนละตัวแปร ลองคลีนโค้ดดูนะครับ
var selectedSum = isX? this.sumOfX : this.sumOfO;
var selectedturnNumber = isX? this.xTurnNumber : this.oTurnNumber;
selectedSum[0] += row;
selectedSum[1] += column;
selectedturnNumber++;